src/scheduler/misc.cpp

```diff
@@ -489,7 +491,31 @@ enum match_string_array_ret match_string_array(const char * const *strarr1, cons
 
 	return SA_NO_MATCH;
 }
+// overloaded
+enum match_string_array_ret match_string_array(const string_vector &strarr1, const string_vector &strarr2)
+{
+	unsigned int match = 0;
+
+	if (strarr1.empty() || strarr2.empty())
+		return SA_NO_MATCH;
+
+	for (auto &str1: strarr1) {
+		if (std::find(strarr2.begin(), strarr2.end(), str1) != strarr2.end())
+			match++;
+	}
+
+	/* i is the length of strarr1 since we just looped through the whole array */
+	if (match == strarr1.size() && match == strarr2.size())
+		return SA_FULL_MATCH;
+
+	if (match == strarr1.size() || match == strarr2.size())
+		return SA_SUB_MATCH;
+
+	if (match)
+		return SA_PARTIAL_MATCH;
 
+	return SA_NO_MATCH;
+}
```

Comment:
If we only ever search for full matches, and you are writing this function over again, consider just shrinking both functions to search for a full match.